src/qt/walletmodel.cpp

```diff
@@ -120,12 +123,18 @@ void WalletModel::pollBalanceChanged()
 
 void WalletModel::checkBalanceChanged(const interfaces::WalletBalances& new_balances)
 {
-    if(new_balances.balanceChanged(m_cached_balances)) {
+    const auto& cached_balance = getCachedBalance();
+    if (new_balances.balanceChanged(cached_balance)) {
```

Comment:
This was because of https://github.com/bitcoin-core/gui/pull/598#discussion_r868204278.

Before I remove that commit, `getCachedBalance` was guarded by a mutex. Now is not needed anymore 👍🏼.